src/main/java/ch/njol/skript/structures/StructVariables.java

```diff
@@ -227,13 +231,15 @@ public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResu
 			}
 			variables.add(new NonNullPair<>(name, o));
 		}
-		getParser().getCurrentScript().addData(new DefaultVariables(variables));
+		script.addData(new DefaultVariables(variables)); // we replace the previous entry
 		return true;
 	}
 
 	@Override
 	public boolean load() {
 		DefaultVariables data = getParser().getCurrentScript().getData(DefaultVariables.class);
+		if (data == null) // band-aid fix for this section's behaviour being handled by a previous section
```

Comment:
if data is null here, then something has gone really wrong

We should prefer an assertion or throw an error - or do nothing at all because there's no way it could be null unless something (e.g. an addon) has dangerously manipulated things (which could happen anywhere)

Reply:
I didn't know if there was a situation where one could load and unload before another loads so I thought I'd better make sure 😬

Reply:
ah it shouldn't. if this did somehow happen we definitely need some sort of feedback (whether that's us checking and throwing an error or allowing the code to error)